### ISSUE TITLE
.evalSync() server error catch

### ISSUE THREAD

**Issue by arunoda** [arunoda/laika#10]
Make it possible to use following

``` js
assert.throws(function() {
    server.evalSync(function() {

        throw new Error('fail');
        emit('return');

    });
});
```

**Comment by arunoda** [comment]
Now we can catch errors in the evalSync. But it is not possible to use `assert.thorws()` since `evalSync` needs to be fired from the main callback.

So you have to write it with try-catch block

``` js
try{
  server.evalSync(function() {
    throw new Error('hoiya');
    emit('return');
  }); 
  assert.fail('should throw an error');
} catch(ex) {
  assert.ok(ex.message.match(/hoiya/));
  done();
}
```

Fixed in: a55897ea36c295f467868afbd10b69215b6d09ca (laika version: 0.1.29)

**Comment by zvictor** [comment]
**Awesome!** But it is not perfect yet, I found a bug =/

It just works for me if I do not run any `evalSync` before the `try` clause.
Otherwise, we are gonna have an error. Code:

``` js
server.evalSync(function() {
  emit('return', false);
});

try{
  server.evalSync(function() {
    throw new Error('hoiya');
    emit('return');
  });
  assert.fail('should throw an error');
} catch(ex) {
  assert.ok(ex.message.match(/hoiya/));
  done();
}
```

output:

``` haskell
Uncaught Error: Future resolved more than once
 at Object.Future.return (/usr/lib/node_modules/laika/node_modules/fibers/future.js:154:10)
 at ServerConnector.<anonymous> (/usr/lib/node_modules/laika/lib/connectors/server.js:66:20)
 at ServerConnector.EventEmitter.emit (events.js:117:20)
 at DtfParser.<anonymous> (/usr/lib/node_modules/laika/lib/connectors/server.js:26:12)
 at DtfParser.EventEmitter.emit (events.js:95:17)
 at /usr/lib/node_modules/laika/lib/dtf_parser.js:24:12
 at Array.forEach (native)
 at DtfParser.parse (/usr/lib/node_modules/laika/lib/dtf_parser.js:22:9)
 at Socket.onClientData (/usr/lib/node_modules/laika/lib/connectors/server.js:17:12)
 at Socket.EventEmitter.emit (events.js:95:17)
```

**Comment by arunoda** [comment]
Yes. That was a huge mistake myself.
I fixed it.

But I encounter some other edge case with try catch.
Now errors we are not capturing will cause laika to exits immediately and not mocha to capture them.
That is because the how fibers work. BTW: I'm not a fiber expert :)  

So I fight with this. And the way all work is as below.

``` js
test('throw error on evalSync', function(done, server, client) {

  var error = server.evalSyncExpcectError('I want some error', function() {
    throw new Error('hoiya');
    emit('return');
  });
  assert.ok(error.message.match(/hoiya/));

  done();
});
```

`evalSyncExpcectError` first errorMessage argument is optional

This is kind a weird, but it is the only working solution I could found.
I open to change the method name :)

This is published as `laika` version 0.2.1

> See Commit: b30cda10a2156e90f6890579ba90a0cb9e0e05d6
> Some example tests: http://goo.gl/JRcq8

**Comment by zvictor** [comment]
that is okay for now.
I am just worried about the name <i>Exp</i>**C**<i>ect</i>. Isn't it a typo?

**Comment by arunoda** [comment]
Yes. It is typo in the code. Will fix that now.

**Comment by arunoda** [comment]
fixed in 0.2.2
